spec/views/content/edit.html.erb_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+RSpec.describe "content/edit.html.erb" do
+  describe "edit document form" do
+    it "should render correctly for every document type" do
+      document_types_we_manage = DocumentType.all.filter { |dt| dt.managed_elsewhere.blank? }
+      document_types_we_manage.each do |document_type|
+        edition = create(:edition, document_type: document_type)
+        assign(:edition, edition)
+        assign(:revision, edition.revision)
+        render
+
+        document_type.contents.each do |field|
+          expect(response).to render_template("content/edit/_#{field.id}_field")
```

Comment:
This doesn't feel like something we want to add as Rails has moved in the direction away from this. See: https://github.com/rails/rails/issues/18950 for more. Also seems a massive shame to add an extra dependency for ensuring coverage.

Reply:
Agreed it's a shame to introduce a dependency, and that testing which file is rendered feels overly tightly coupled to implementation detail.

I considered searching for specific strings in the rendered output, corresponding to each input field, but the ids don't correspond to anything in the HTML form, so it would have to be a hardcoded lookup. Alternatively I could `expect(response.scan("<textarea")).to have_exactly(3).items` - but that relies on us using only textarea for our inputs 😕

Reply:
Have talked IRL with Ben and settled on a schema validation approach - it's more expressive than attempting to validate fields in pure RSpec.